src/melbalabs/summarize_consumes/main.py

```diff
@@ -1385,12 +1385,105 @@ def add_extra_attacks(self, howmany, source, name):
         self.counts_extra_attacks[source][name] += howmany
 
 
+class Currency(int):
+    string_pattern = r'((?P<gold>\d+)g)?\s?((?P<silver>\d+)s)?\s?((?P<copper>\d+)c)?'
 
-class PrintConsumables:
-    def __init__(self, player, pricedb, death_count):
+    def __new__(cls, value, *args, **kwargs) -> Self:
+        if isinstance(value, str):
+            return cls.from_string(value)
+        return super().__new__(cls, value, *args, **kwargs)
+
+    def __add__(self, other) -> Self:
+        return Currency(super().__add__(other))
+
+    def __sub__(self, other) -> Self:
+        return Currency(super().__sub__(other))
+
+    def __mul__(self, other) -> Self:
+        return Currency(super().__mul__(other))
+
+    def __truediv__(self, other) -> Self:
+        return Currency(super().__truediv__(other))
+
+    def __mod__(self, other) -> Self:
+        return Currency(super().__mod__(other))
+
+    @classmethod
+    def from_string(cls, s: str) -> Self:
+        m = re.search(cls.string_pattern, s)
+        if not any(m.groupdict().values()):
+            raise ValueError(f"Invalid currency string format: {s}")
+        return cls(sum([int(m.group('gold') or 0) * 10000,
+                        int(m.group('silver') or 0) * 100,
+                        int(m.group('copper') or 0)]))
+
+    def to_string(self, short: bool=False) -> str:
+        if short:
+            return f"{int(self) / 10000.0:.1f}g"
+
+        value = int(self)
+        copper = value % 100
+        value = value // 100
+        silver = value % 100
+        value = value // 100
+        gold = value
+        s = ''
+
+        first = True
+        for amount, suffix in zip([gold, silver, copper], 'gsc'):
+            if amount:
+                if not first: s+= ' '
+                else: first = False
+                s += f'{amount}{suffix}'
+        return s
+
+
+class ConsumablesAccumulator:
+    def __init__(self, player, pricedb, death_count) -> Self:
         self.player = player
         self.pricedb = pricedb
         self.death_count = death_count
+        self.data = {}
+
+    def get_consumable_price(self, consumable):
```

Comment:
no need to do anything now, just a design comment

making this a member function of this class means some random code now has to instantiate a ConsumablesAccumulator just to get the price of a consumable, which makes 0 sense. 
Another symptom is the function doesn't use the self.data attribute, which is the main purpose of this class.